modules/openapi-generator/src/main/resources/python/python-experimental/model_templates/model_composed.mustache

```diff
@@ -15,6 +15,12 @@ class {{unescapedDescription}}(ModelComposed):
 
 {{> python-experimental/model_templates/classvars }}
 
+{{#oneOf}}
+{{#-first}}
+    attribute_map = {}
```

Comment:
The correct solution would include all properties defined directly on the model and not in any `oneOf` one.

Reply:
I am a little concerned that this implementation turns off type checking in the composed schema. This is probably okay, because when we assign the variables to the oneOf/anyOf class instances, the type is also checked there.

Can't this happen for all of these cases for required + optional variables:
1. oneOf schemas have a variable with conflicting type
2. anyOf schemas have a variable of conflicting type
3. oneOf + anyOf schemas have a variable of conflicting type?

You have also revealed that the docstring for this function is incorrect for these 1-3 use cases.
Python-experimental has previously addressed issues with composed schema inputs and tweaking them in the past with the method [addNullDefaultToOneOfAnyOfReqProps](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PythonClientExperimentalCodegen.java#L738). It goes through and sets composed schema required inputs to have a default null value in the composed schema init signature. This makes those inputs named arguments and makes them optional for users to pass them in. This is because:
we can have 2 oneOf schemas, schemaA, and schemaB each with 1 required variable, varA, varB. If we are only making an instance of schemaA, then varB is not required. So we set them all to null, and when we make an instance of schemaA, it requires that we pass in varA.

How about we do some similar tweaking at the composed schema variable level here?
We could
- look for variable type collisions in the oneOf/anyOf variables of a composed schema
- update their data types at the composed schema level to be a tuple of the allowed types
- this lets us keep using the mustache template as-is
- this keeps our documentation up to date

What do you think?

Reply:
> 1. oneOf schemas have a variable with conflicting type

This can happen and we should support it.

> 2. anyOf schemas have a variable of conflicting type

It can happen. I am not sure how it should be resolved when a value matches multiple conflicting types.

> 3. Does our generator allow type collisions between variables in (allOf + oneOf) or (allOf + anyOf)? It should not, that violates the openapi spec.

You are right that *It should not*.

I am not sure about the solution though. I would completely remove validation on base schema for properties defined in *X*Of models.

---
I agree that we should not merge this mustache template patch. I was looking for a quick solution that would unblock me so I can at least generate code that can deserialize server response without raising an exception.

Reply:
> Does our generator allow type collisions between variables in (allOf + oneOf) or (allOf + anyOf)? It should not, that violates the openapi spec.

One can intentionally or accidentally write a OAS document which is 1) syntactically compliant with the JSON schema specification and 2) will not match *any* input payload. It's an interesting question whether it's even possible to analyze a schema and conclude whether there is at least one matching input data. 
IMO this should be handled with the `--validation` argument, and the code generator should still accept the OAS document. I.e. you want to know if a OAS doc is syntactically valid.

Consider `pattern`. There is an infinite number of regex that will not match any input string, including the empty string. Here is one simple example:
```yaml
type: string
pattern: .*
```